sql: avoid flake in scan_test

jordanlewis opened the pull request:
The new sqlliveness package doesn't get along well with the kv batch
size "atomic", so turn down its check frequency in the test. Possibly we
should make the "atomic" actually atomic, but I'm not sure about the
runtime performance change.

Fixes #52683

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52769)
<!-- Reviewable:end -->

asubiotto reviewed (approved):
An atomic should be fine in the grand scheme of things, but this solution <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52769#-:-MEcA6tD7DF_wGSSImJm:buftqey)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

jbowens reviewed (commented):
Is this good to merge @jordanlewis? Seeing more of this failure today: https://github.com/cockroachdb/cockroach/issues/52683

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52769#-:-MEyJ-Vj1hrGzOCwfYmk:b-8tuhwf)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @jordanlewis)

___
*[pkg/sql/scan_test.go, line 175 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52769#-MEyIydM808bTcrjLPRv:-MEyIydM808bTcrjLPRw:bn7alyj) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd9632917c4351505d5378753247cd766180a3b0/pkg/sql/scan_test.go#L175)):*
> ```Go
> 
> 	for _, batch := range batchSizes {
> 		row.TestingSetKVBatchSize(int64(batch))
> ```

accident?


<!-- Sent from Reviewable.io -->

jordanlewis committed:
sql: avoid flake in scan_test

The new sqlliveness package doesn't get along well with the kv batch
size "atomic", so turn down its check frequency in the test. Possibly we
should make the "atomic" actually atomic, but I'm not sure about the
runtime performance change.

Fixes #52683

Release note: None

asubiotto commented:
Jordan's out so merging this on his behalf

bors r=asubiotto

jbowens commented:
Thanks @asubiotto
